packages/core/src/server.ts

```diff
@@ -252,10 +257,12 @@ export class Controller extends API {
         const { auth } = this._requireAuth();
         const authenticator = new Authenticator({ type, purposes, device });
         await authenticator.init();
+        const originalAuth = deepClone(auth);
         const provider = this._getAuthServer(type);
         const responseData = await provider.initAuthenticator(authenticator, auth, data);
         auth.authenticators.push(authenticator);
         await this.storage.save(auth);
+        this.auditor.log("auth", auth.id, auth.accountId!, "updated", originalAuth, auth);
```

Comment:
So, we can do this, or, an alternative that's not uncommon, would be to pass the `auditor` to the `storage`, and every time we run `storage.save()`, we'd try to run `auditor.log` (knowing that some changes don't need to be individually audited -- sessions and provisioning, for example), but I wasn't sure if you liked that.